storage/scratchpad/src/sparse_merkle/ancestors.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::{sparse_merkle::metrics::TIMER, SparseMerkleTree};
+use aptos_crypto::hash::CryptoHash;
+use aptos_drop_helper::async_drop_queue::AsyncDropQueue;
+use aptos_infallible::Mutex;
+use aptos_logger::prelude::*;
+use aptos_metrics_core::TimerHelper;
+use std::{
+    collections::VecDeque,
+    sync::{Arc, MutexGuard},
+    time::Duration,
+};
+
+type Result<V> = std::result::Result<V, Error>;
+
+/// Keep the oldest SMTs in a central place so that:
+/// 1. elsewhere, dropping the SMT will be fast since this indirectly hols a ref to every tree in
+///    the entire forest.
+/// 2. this must be invoked somewhere in the critical path so that it can provide some
+///    back pressure to slow things down to prevent from leaking memory
+#[derive(Clone, Debug)]
+pub struct SmtAncestors<V: Clone + Send + Sync + 'static> {
```

Comment:
nit: remove Clone

Reply:
Clone is required down the stack from the updater. I'd follow up later on this in favor of time. 😢